td-shim-tools/src/enroller.rs

```diff
@@ -197,7 +206,9 @@ pub fn create_key_file(key_file: &str, hash_alg: &str) -> io::Result<FirmwareRaw
                         "unsupported Named Curve",
                     ));
                 }
-                if key.subject_public_key.as_bytes()[0] != 0x04 {
+                if key.subject_public_key.as_bytes()[0] != 0x04
```

Comment:
what is 0x04 ?

Reply:
0x04 means the key is uncompressed

Reply:
I dont see there is such definition in https://github.com/confidential-containers/td-shim/blob/main/doc/secure_boot.md.

Reply:
It does not exist in the td shim binary, the key filled into `Signature Block` is 48 + 48 bytes.

Reply:
Please add detail description on the expectation of the key data.

Reply:
Also, what sort of compression is being used here? Could that ever change? If so, a `const` naming the type and version of compression algorithm represented by `0x04` would be useful.

Reply:
The compressed form is shorter than the uncompressed one, for ECDSA P384, the compressed public key is encoded in 49 bytes and uncompressed public key is endcoded in 97 bytes (both include a one-byte prefix), the compression form leaves out half of the coordinate value which can be calculated according to the fixed curve and the prefix. According to the [Elliptic Curve Cryptography Subject Public Key Information](https://www.rfc-editor.org/rfc/rfc5480#section-2.2), the uncompressed form is indicated by 0x04 and the compressed form is indicated by either 0x02 or 0x03, and it will not change.

Reply:
I think to support uncompressed version only is OK.
My point is:
1) Please define 0x04 as a meaningful macro/const value so that people know what is is. A magic number is not acceptable.
2) Please define ECDSA-384 key as 96 byte. If the extra byte is 0x4 as uncompressed indicator, it should be parsed separately. Defining ECDSA-384 key as 97 bytes is very confusing.